wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DtoConverter.java

```diff
@@ -187,10 +189,30 @@ private static DevfileCommandDto asDto(
       org.eclipse.che.api.core.model.workspace.devfile.Command command) {
     List<DevfileActionDto> actions =
         command.getActions().stream().map(DtoConverter::asDto).collect(toList());
-    return newDto(DevfileCommandDto.class)
-        .withName(command.getName())
-        .withActions(actions)
-        .withAttributes(command.getAttributes());
+
+    DevfileCommandDto commandDto =
+        newDto(DevfileCommandDto.class)
+            .withName(command.getName())
+            .withActions(actions)
+            .withAttributes(command.getAttributes());
+
+    if (command.getPreviewUrl() != null) {
+      commandDto.setPreviewUrl(asDto(command.getPreviewUrl()));
+    }
+    return commandDto;
+  }
+
+  private static PreviewUrlDto asDto(PreviewUrl previewUrl) {
+    final PreviewUrlDto previewUrlDto = newDto(PreviewUrlDto.class);
+    if (previewUrl != null) {
```

Comment:
We're representing a `null` input with "empty" instance. While that might be technically OK for DTOs, it makes me wonder whether we correctly distinguish between the two when constructing the servers? Like is it possible that the provisioning will create a service and ingress with `null` path on port `0`?

Reply:
`null` path is valid. And port `0` is technically also valid. So as it probably does not make much sense, it is possible. There is currently only check for `null` when constructing servers.